l10n_it_sdi_channel/models/sdi.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Sergio Corato (https://efatto.it)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, _, exceptions
+from odoo.addons.base.ir.ir_mail_server import extract_rfc2822_addresses
+
+SDI_CHANNELS = [
+    ('pec', 'PEC'),
+    # ('web', 'Web service'), # not implemented
+    # ('sftp', 'SFTP'), # not implemented
+]
+
+
+class SdiChannel(models.Model):
+    _name = "sdi.channel"
+    _description = "SdI channel"
+
+    name = fields.Char(string='Name', required=True)
+    company_id = fields.Many2one(
+        'res.company', string='Company', required=True,
+        default=lambda self:
+        self.env['res.company']._company_default_get('sdi.channel'))
+    channel_type = fields.Selection(
+        string='SdI channel type', selection=SDI_CHANNELS, required=True,
+        help='PEC is the only implemented channel in this module. Other '
+             'channels (Web, Sftp) could be provided by external modules.')
+    pec_server_id = fields.Many2one(
+        'ir.mail_server', string='Pec mail server', required=False,
+        domain=[('is_fatturapa_pec', '=', True)])
+    email_from_for_fatturaPA = fields.Char(
```

Comment:
questo campo credo sia inutile, la mail pec di spedizione non può essere diversa da quella del server pec collegato, provvedo quindi a eliminarlo

Reply:
leggendo però [qui](https://stackoverflow.com/questions/37322897/are-imap-smtp-credentials-always-the-same-as-the-users-login-credentials) potrebbe essere diverso l'utente dalla mail gestita, per cui lo gestisco a livello di server pec